flatland_server/src/yaml_preprocessor.cpp

```diff
@@ -0,0 +1,214 @@
+ * @brief	 Yaml preprocessor using Lua
+ * @author Joseph Duchesne
+ *
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2017, Avidbots Corp.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *      copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the Avidbots Corp. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "flatland_server/yaml_preprocessor.h"
+#include <ros/ros.h>
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/lexical_cast.hpp>
+
+#include <cstdlib>
+#include <cstring>
+
+namespace flatland_server {
+
+void YamlPreprocessor::Parse(YAML::Node &node) {
+  YamlPreprocessor::ProcessNodes(node);
+}
+
+void YamlPreprocessor::ProcessNodes(YAML::Node &node) {
+  switch (node.Type()) {
+    case YAML::NodeType::Sequence:
+      for (YAML::Node child : node) {
+        YamlPreprocessor::ProcessNodes(child);
+      }
+      break;
+    case YAML::NodeType::Map:
+      for (YAML::iterator it = node.begin(); it != node.end(); ++it) {
+        YamlPreprocessor::ProcessNodes(it->second);
+      }
+      break;
+    case YAML::NodeType::Scalar:
+      if (node.as<std::string>().compare(0, 5, "$eval") == 0) {
+        ProcessScalarNode(node);
+      }
+      break;
+  }
+}
+
+void YamlPreprocessor::ProcessScalarNode(YAML::Node &node) {
+  std::string value = node.as<std::string>().substr(5);  // omit the $parse
+  boost::algorithm::trim(value);                         // trim whitespace
+  ROS_INFO_STREAM("Attempting to parse lua " << value);
+
+  try {
+    if (value.find("return ") ==
```

Comment:
The scope of this try{} block can be reduced by moving these string manipulations out.